etcdutl/snapshot/v3_snapshot.go

```diff
@@ -311,6 +327,73 @@ func (s *v3Manager) saveDB() error {
 	return nil
 }
 
+// maybeModifyLatestRevision can increase the latest revision by the given amount and sets the scheduled compaction
+// to that revision so that the server will compact the database up to that revision.
+func (s *v3Manager) maybeModifyLatestRevision(markCompacted bool, bumpAmount int64) error {
+	be := backend.NewDefaultBackend(s.lg, s.outDbPath())
+	defer be.Close()
+
+	tx := be.BatchTx()
+	tx.LockOutsideApply()
+	defer tx.Unlock()
+
+	latest, err := s.unsafeGetLatestRevision(tx)
+	if err != nil {
+		return err
+	}
+
+	if bumpAmount > 0 {
+		latest = s.unsafeBumpRevision(tx, latest, bumpAmount)
+	}
+
+	if markCompacted {
+		s.unsafeMarkRevisionCompacted(tx, latest)
+	}
+
+	be.ForceCommit()
+
+	return nil
+}
+
+func (s *v3Manager) unsafeBumpRevision(tx backend.BatchTx, latest revision, amount int64) revision {
+	s.lg.Info(
+		"bumping latest revision",
+		zap.Int64("latest-revision", latest.main),
+		zap.Int64("bump-amount", amount),
+		zap.Int64("new-latest-revision", latest.main+amount),
+	)
+
+	latest.main += amount
+	k := [17]byte{}
+	revToBytes(k[:], latest)
+	tx.UnsafePut(schema.Key, k[:], []byte{})
+
+	return latest
+}
+
+func (s *v3Manager) unsafeMarkRevisionCompacted(tx backend.BatchTx, latest revision) {
+	s.lg.Info(
+		"marking revision compacted",
+		zap.Int64("revision", latest.main),
+	)
+
+	mvcc.UnsafeSetScheduledCompact(tx, latest.main)
```

Comment:
Please make sure to get semantics of subrevision right. I think to mark revision as compacted (X,Y) we put (X+1, 0) as compacted.

Reply:
Do you mean that you would put an additional revision at X+1 then pass that into SetScheduledCompact or pass in X+1 to SetScheduledCompact? The latter at least doesn't work and causes a future revision error on startup.

Reply:
Wasn't this about watch ?